tl/features/cell_context_matches.py

```diff
@@ -233,14 +319,61 @@ def compute_context_similarity(self,
 
         return result
 
-    def computes_string_similarity(self, values: List[str], col2_string: str) -> Tuple[float, str]:
-        return 0.0, "best_matched_string"
+    def preprocess(self, word: str) -> list:
+        word = word.lower()
+        preprocessed_word = re.sub(r'[^\w\s]', '', word)
+        preprocessed_word = preprocessed_word.split(" ")
+        return preprocessed_word
+
+    @staticmethod
+    def quantity_score(quantity_1: float, quantity_2: float) -> float:
+        """
+        Purpose: Calculates the score between two quantities by taking the absolute difference between them and
+        dividing by the max of both.
+        It is then subtracted from 1.
+        Returns: score
+        """
+        if quantity_1 == 0.0 and quantity_2 == 0.0:
+            return 1
+        return 1 - (abs(quantity_1 - quantity_2) / max(abs(quantity_1), abs(quantity_2)))
+
+    def computes_string_similarity(self, values: List[str], col2_string_list: List[str], col2_string_datatype: List[str], values_type: str) -> Tuple[float, str]:
+        # Estimate if there's a match between the types.
+        if values_type not in col2_string_datatype:
+            return 0.0, ""
+        max_sim = 0.0
+        best_matched = ""
+        if values_type == "i":
+            for col_split in col2_string_list:
+                for val in values:
+                    sim = similarity.hybrid.symmetric_monge_elkan_similarity(self.preprocess(val),
```

Comment:
i think we discussed a long time ago, that the rltk takes a threshold which should make this code faster, where is that parameter?

Reply:
I did not add the parameter for this section as until then I had not started taking threshold parameters in the code.